k2/csrc/fsa_utils.cu

```diff
@@ -2096,4 +2097,395 @@ void FixNumStates(FsaVec *fsas) {
   fsas->shape = RemoveSomeEmptyLists(fsas->shape, 1, renumber_states);
 }
 
+template <typename FloatType>
+Array1<FloatType> GetArcCdf(FsaOrVec &fsas,
+                            Array1<FloatType> &arc_post) {
+  K2_CHECK_GE(fsas.NumAxes(), 2);
+  K2_CHECK_LE(fsas.NumAxes(), 3);
+  int32_t state_axis = (fsas.NumAxes() == 3 ? 1 : 0);
+  ContextPtr c = fsas.Context();
+  int32_t num_states = fsas.TotSize(state_axis),
+      num_arcs = fsas.NumElements();
+  Array1<FloatType> state_post(c, num_states);
+  // Use lowest() instead of -infinity() to initialize the sum, to avoid
+  // -infinity - (-infinity) = NaN; but actually this shouldn't really be an
+  // issue as we shouldn't be taking paths with scores equal to -infinity.
+  LogSumPerSublist(fsas,
+                   std::numeric_limits<FloatType>::lowest(),
```

Comment:
just noted we can also here init with `-infinity` and check `state_post_data != -inifinity` below, I have used this approach for ComputeScores/ArcPost related functions, it works well. But you version would also work well I believe.